File: backend/src/main/java/ca/metricalsky/winston/entity/SettingsEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "settings")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SettingsEntity {

    @Id
    @Column(name = "setting_name")
    private String name;

    @Basic
    @Column(name = "setting_value")
    private String value;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/SettingsEntity.kt
package ca.metricalsky.winston.database.entity

import jakarta.persistence.Basic
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "settings")
class SettingsEntity (

    @Id
    @Column(name = "setting_name")
    var name: String,

    @Basic
    @Column(name = "setting_value")
    var value: String
)
